docs/DeveloperGuide.md

```diff
@@ -1,38 +1,461 @@
 
-## Acknowledgements
+## **Acknowledgements**
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+* The formatting for the developer guide is inspired by [AB-3](https://se-education.org/addressbook-level3/DeveloperGuide.html).
 
-## Design & implementation
+Third-party libraries:
+* [OpenCSV](https://opencsv.sourceforge.net/) - This package is licensed under [Apache2](https://opencsv.sourceforge.net/licenses.html), which is a business-friendly open-source software license.
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+--------------------------------------------------------------------------------------------------------------------
 
-## Product scope
-### Target user profile
+## **Design**
 
-{Describe the target user profile}
+<div markdown="span" class="alert alert-primary">
+:bulb: **Tip:** The `.puml` files used to create diagrams in this document `docs/diagrams` folder. Refer to the [_PlantUML Tutorial_ at se-edu/guides](https://se-education.org/guides/tutorials/plantUml.html) to learn how to create and edit diagrams.
+</div>
 
-### Value proposition
+### Architecture
+<img src="images/ArchitectureDiagram.png" alt=""/>
 
-{Describe the value proposition: what problem does it solve?}
+The ***Architecture Diagram*** given above explains the high-level design of the App.
+
+Given below is a quick overview of main components and how they interact with each other.
+
+**Main components of the architecture**
+
+[**`StockPal`**](https://github.com/AY2324S2-CS2113T-T09-3/tp/blob/master/src/main/java/seedu/stockpal/StockPal.java) is in charge of the app launch and shut down.
+
+The bulk of the app's work is done by the following five components:
+
+* [**`UI`**](#ui-component): The UI of the App.
+* [**`Parser`**](#parser-component): Parses user input into respective commands.
+* [**`Commands`**](#commands-component): The command executor.
+* [**`Data`**](#data-component): Holds the data of the App in memory.
+* [**`Storage`**](#storage-component): Reads data from, and writes data to, the hard disk.
+
+[**`Commons`**](#common-classes) represents a collection of classes used by multiple components above.
+[**`Exceptions`**](#exceptions-classes) represents a collection of exceptions used by multiple components above.
+
+**How the architecture components interact with each other**
+
+The *Sequence Diagram* below shows how the components interact with each other for the scenario where the user issues the command `delete 1`.
+
+<img src="images/ArchitectureSequenceDiagram.png" alt=""/>
+
+The sections below give more details of each component.
+
+### UI component
+The **API** of this component is specified in [`Ui.java`](https://github.com/AY2324S2-CS2113T-T09-3/tp/tree/master/src/main/java/seedu/stockpal/ui/Ui.java)
+
+![Structure of the UI Component](images/UiClassDiagram.png)
```

Comment:
Image is not showing up correctly probably because the name of the png was changed from 'uiClassDiagram' to just 'Ui' 
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/110718665/c08f4f1b-4fbd-4e76-a228-3c5ba7f0f17f)